add-on/panels/js/panel.jsx

```diff
@@ -755,32 +770,11 @@ loop.panel = (function(_, mozL10n) {
         return this._renderLoadingRoomsView();
       }
 
-      if (!this.state.rooms.length) {
-        return this._renderNoRoomsView();
-      }
-
       return (
         <div className="rooms">
           {this._renderNewRoomButton()}
-          <h1>{mozL10n.get(this.state.openedRoom === null ?
-                "rooms_list_recently_browsed2" :
-                "rooms_list_currently_browsing2")}</h1>
-          <div className="room-list">{
-            this.state.rooms.map(function(room) {
-              if (this.state.openedRoom !== null &&
-                room.roomToken !== this.state.openedRoom) {
-                return null;
-              }
-
-              return (
-                <RoomEntry
-                  dispatcher={this.props.dispatcher}
-                  isOpenedRoom={room.roomToken === this.state.openedRoom}
-                  key={room.roomToken}
-                  room={room} />
-              );
-            }, this)
-          }</div>
+          {this._renderRoomsTitle()}
+          {this._renderRoomList()}
```

Comment:
Originally the mark-up was in the render and would break as soon as I added conditionals and moved elements.  Created functions as they are allowed in the render and not so much the local variables (var localVar) in the render.